### PR TITLE
#43 Add docker sbt capability

### PR DESCRIPTION
**Pull request by Zejnilovic**
to run 

```shell
$ docker buildx build -t absaoss/atum-service:latest . 
```

### CODE REVIEW

**Comment by lsulak** [comment]
We can 'steal' stuff from Login Service. Not sure if you want to carry on with this PR? But I guess not, since a lot was developed since you started this PR, i.e. for the last 3 months

File: server/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+# LS docker file
```
Comment (lsulak) [inline server/Dockerfile:17]:
```suggestion
# Atum Service docker file
```

File: server/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+# In case you build the Dockerfile from another location than the default 'service' dir, provide a prefix to reach it
+ARG AS_PREFIX=.
+
+LABEL org.opencontainers.image.authors="ABSA"
+
+# Copy Spring application properties
+COPY $CONFIG /opt/application.yaml
+
+# deploy as root application in tomcat
+COPY ${AS_PREFIX}/target/scala-2.12/*.war /usr/local/tomcat/webapps/ROOT.war
```
Comment (lsulak) [inline server/Dockerfile:57]: Perhaps `scala-2.12` could be parametrized/extracted into some constant?

**Review by lsulak — COMMENTED** [review]
LGTM, I haven't tried but it looks almost the same as in LS so I wouldn't anticipate any problems.

**Comment by lsulak** [comment]
So this plugin is not gonna be used? `addSbtPlugin("com.github.sbt" % "sbt-native-packager" % "1.9.4")` 

I think that it's still quite cool

**Comment by Zejnilovic** [comment]
> So this plugin is not gonna be used? `addSbtPlugin("com.github.sbt" % "sbt-native-packager" % "1.9.4")`
> 
> I think that it's still quite cool

I agree; it is cool. But why would I keep it? What is the usecase?

**Comment by lsulak** [comment]
> > So this plugin is not gonna be used? `addSbtPlugin("com.github.sbt" % "sbt-native-packager" % "1.9.4")`
> > I think that it's still quite cool
> 
> I agree; it is cool. But why would I keep it? What is the usecase?

just for building the docker easily via sbt, i.e. convenience?

File: server/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+# Version of scala that server was built against
+ARG SCALA_VERSION=2.12
+
+LABEL org.opencontainers.image.authors="ABSA"
+
+# Copy Spring application properties
+COPY $CONFIG /opt/application.yaml
+
+# deploy as root application in tomcat
+COPY ${AS_PREFIX}/target/scala-${SCALA_VERSION}/*.war /usr/local/tomcat/webapps/ROOT.war
```
Comment (lsulak) [inline server/Dockerfile:59]: this doesn't work, for some reason the actual path where this war is being put is `jvm-*/` and not `scala-*/` so it needs to be changed in `build.sbt` or here. I think that in LSD guys didn't use plugin `sbt-assembly` and their build looks simpler

```
 => [base 2/6] COPY ./src/main/resources/application.properties /opt/application.yaml                                                 0.0s
 => ERROR [base 3/6] COPY ./target/scala-2.12/*.war /usr/local/tomcat/webapps/ROOT.war                                                0.0s
------
 > [base 3/6] COPY ./target/scala-2.12/*.war /usr/local/tomcat/webapps/ROOT.war:
------
Dockerfile:59
--------------------
  57 |
  58 |     # deploy as root application in tomcat
  59 | >>> COPY ${AS_PREFIX}/target/scala-${SCALA_VERSION}/*.war /usr/local/tomcat/webapps/ROOT.war
  60 |
  61 |     ENV SPRING_CONFIG_LOCATION=/opt/application.yaml
--------------------
ERROR: failed to solve: lstat /var/lib/docker/tmp/buildkit-mount1632225145/target/scala-2.12: no such file or directory
```

Comment (Zejnilovic) [reply server/Dockerfile:59]: I have it working and tested, and `war` on my local machine is in the scala folder. 

image

What command are you running?

Comment (Zejnilovic) [reply server/Dockerfile]: Pushed a fix, I had cached some stuff and I was still getting my war from `scala-2.12`.

**Pull request merged**